locking/locks.go

```diff
@@ -225,33 +249,101 @@ func (c *Client) lockIdFromPath(path string) (string, error) {
 	}
 }
 
+// We want to use a single cache file for integrity, but to make it easy to
+// list all locks, prefix the id->path map in a way we can identify (something
+// that won't be in a path)
+const idKeyPrefix = "*id*://"
+
+func (c *Client) encodeIdKey(id string) string {
+	// Safety against accidents
+	if !c.isIdKey(id) {
+		return idKeyPrefix + id
+	}
+	return id
+}
+func (c *Client) decodeIdKey(key string) string {
+	// Safety against accidents
+	if c.isIdKey(key) {
+		return key[len(idKeyPrefix):]
+	}
+	return key
+}
+func (c *Client) isIdKey(key string) bool {
+	return strings.HasPrefix(key, idKeyPrefix)
+}
+
 // Cache a successful lock for faster local lookup later
-func (c *Client) cacheLock(filePath, id string) error {
-	// TODO
+func (c *Client) cacheLock(l Lock) error {
+	// Store reference in both directions
+	// Path -> Lock
+	c.cache.Set(l.Path, &l)
+	// EncodedId -> Lock (encoded so we can easily identify)
+	c.cache.Set(c.encodeIdKey(l.Id), &l)
 	return nil
 }
 
 // Remove a cached lock by path becuase it's been relinquished
-func (c *Client) cacheUnlock(filePath string) error {
-	// TODO
+func (c *Client) cacheUnlockByPath(filePath string) error {
+	ilock := c.cache.Get(filePath)
+	if ilock != nil {
+		lock := ilock.(*Lock)
```

Comment:
Not that this is likely behavior, but I think that using the two-tuple variant of this cast would be safer in that it'd prevent an unlikely panic:

```go
func (c *Client) cacheUnlockByPath(filePath string) error {
        ilock := c.cache.Get(filePath)
        if ilock == nil { // <- needed to prevent `(*Lock)(nil)`
                return nil
        }
        if l, ok := ilock.(*Lock); ok {
                // ...
        }

        return nil
}
```